src/application-navigation/ApplicationNavigation.jsx

```diff
@@ -0,0 +1,192 @@
+  /**
+   * An element to render within the ApplicationNavigation menu, shifted to the drawer at the `medium` and below.
+   */
+  hero: PropTypes.element,
+  /**
+   * An array of configuration objects with information specifying the creation of navigation items. These items
+   * are rendered within the ApplicationNavigation header at larger breakpoints and within the drawer menu at smaller breakpoints.
+   */
+  navigationItems: navigationItemsPropType,
+  /**
+   * A string key representing the currently active navigation item. This value should match one of the item keys provided in the
+   * `navigationItems` array.
+   */
+  activeNavigationItemKey: PropTypes.string,
+  /**
+   * Callback function triggered on Drawer Menu state change
+  */
+  onDrawerMenuStateChange: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of a navigation item.
+   * Ex: `onSelectNavigationItem(String selectedNavigationItemKey, Object metaData)`
+   */
+  onSelectNavigationItem: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Settings utility item.
+   * If `onSelectSettings` is not provided, the Settings utility item will not be rendered.
+   * Ex: `onSelectSettings()`
+   */
+  onSelectSettings: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Help utility item.
+   * If `onSelectHelp` is not provided, the Help utility item will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectHelp: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Logout action button.
+   * If `onSelectLogout` is not provided, the Logout action button will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectLogout: PropTypes.func,
+  /**
+   * An array of configuration objects with information specifying the creation of additional utility menu items.
+   * These items are rendered within the popup utility menu at larger breakpoints and within the drawer menu at smaller breakpoints.
+   */
+  utilityItems: utilityItemsPropType,
+  /**
+   * A function to be executed upon the selection of a custom utility item.
+   * Ex: `onSelectUtilityItem(String selectedUtilityItemKey, Object metaData)`
+   */
+  onSelectUtilityItem: PropTypes.func,
+  /**
+   * A collection of child elements to render within the ApplicationNavigation body.
+   */
+  children: PropTypes.node,
+  /**
+   * Key/Value pairs associating a string key entry to a numerical notification count.
+   */
+  notifications: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
+  /**
+   * By default, the ApplicationNavigation component will resolve any rendered NavigationPrompts prior to
+   * communicating navigation item selections with `onSelectNavigationItem`. If `disablePromptsForNavigationItems`
+   * is provided, no NavigationPrompts are resolved when navigation items are selected.
+   */
+  disablePromptsForNavigationItems: PropTypes.bool,
```

Comment:
When would we want this to be enabled? By the testing framework?

Reply:
By default, ApplicationNavigation assumes that changing primary navigation items is always a destructive action for the page contents. This is how we expect people to consume the component, and this is why we have the prompts enabled by default.

However, depending on how an application is built, changing navigation items may not actually result in data loss. This prop exists to allow consumers the option of disabling the default prompts and handling everything themselves.